### DISCUSSION TITLE
Spark Linker predictions on two PySpark dataframes

### DISCUSSION THREAD

**Discussion by prabh-singh123** [moj-analytical-services/splink#574] [Q&A] [answered]
I estimated model parameters (m & u) on a dataset of 10,000 points and saved the settings dictionary. Then, created a linker object with the same 10,000 points and used the settings dictionary saved before. Everything works fine, and I get 121833 rows as a result of the predict method as shown below. 

![image](https://user-images.githubusercontent.com/105444112/176278308-6d8bc2e0-8c26-4c75-8ba7-271310bf7745.png)

![image](https://user-images.githubusercontent.com/105444112/176279001-ea401ca2-dae9-4c4a-b880-84f82f237e02.png)

Next, I created a new linker object with a sampled dataframe of 1,000 points and used the same saved settings from above. However, when I run the predict method with this new linker object, I still get the exact same results in predictions. 

![image](https://user-images.githubusercontent.com/105444112/176279740-ff051449-415b-4d76-9322-ad0112711982.png)

![image](https://user-images.githubusercontent.com/105444112/176279833-7ed9ee2a-cec0-4923-98c5-df2907aa2444.png)

Is this expected behaviour? How can I get predictions on the new dataframe I just created?

**Comment by RobinL** [comment, marked as answer]
Thanks for the report.  I think this is likely a bug.  Splink makes extensive use of caching for performance reasons so I suspect what's going on here is that the input data has been cached.

In the short term, running the second link after restarting your spark session should fix this.

I've converted this into an issue [here](https://github.com/moj-analytical-services/splink/issues/575)

You're probably aware of this already, but for smaller datasets (less than about a million records), you can use the DuckDB linker, which should be faster, and I don't think you would run into this bug
